ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/BackupRollbackIntegrationTest.java

```diff
@@ -0,0 +1,233 @@
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
+      .around(TEST_HARNESS);
+
+  @ClassRule
+  public static final TemporaryFolder TMP_FOLDER = new TemporaryFolder();
+
+  private static File BACKUP_LOCATION;
+  private static TestKsqlRestApp REST_APP;
+  private String commandTopic;
+  private Path backupFile;
+  private KsqlRestConfig ksqlRestConfig;
+  private KsqlConfig ksqlConfig;
+
+  @BeforeClass
+  public static void classSetUp() throws IOException {
+    BACKUP_LOCATION = TMP_FOLDER.newFolder();
+
+    REST_APP = TestKsqlRestApp
+        .builder(TEST_HARNESS::kafkaBootstrapServers)
+        .withProperty(KSQL_STREAMS_PREFIX + StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1)
+        .withProperty(KSQL_METASTORE_BACKUP_LOCATION, BACKUP_LOCATION.getPath())
+        .build();
+  }
+
+  @Before
+  public void setup() throws IOException {
+    REST_APP.start();
+    ksqlRestConfig = REST_APP.getKsqlRestConfig();
+    ksqlConfig = new KsqlConfig(ksqlRestConfig.getKsqlConfigProperties());
+    commandTopic = ReservedInternalTopics.commandTopic(ksqlConfig);
+    backupFile = Files.list(BACKUP_LOCATION.toPath()).findFirst().get();
+  }
+
+  @After
+  public void teardown() {
+    REST_APP.stop();
+    TEST_HARNESS.deleteTopics(Collections.singletonList(commandTopic));
+  }
+
+  @After
+  public void teardownClass() {
+    TMP_FOLDER.delete();
+  }
+
+  @Test
+  public void shouldBackupAndRestoreCommandTopic() throws Exception {
+    // Given
+    TEST_HARNESS.ensureTopics("topic1", "topic2");
+
+    makeKsqlRequest("CREATE STREAM TOPIC1 (ID INT) "
+        + "WITH (KAFKA_TOPIC='topic1', VALUE_FORMAT='JSON');");
+    makeKsqlRequest("CREATE STREAM TOPIC2 (ID INT) "
+        + "WITH (KAFKA_TOPIC='topic2', VALUE_FORMAT='JSON');");
+    makeKsqlRequest("CREATE STREAM stream1 AS SELECT * FROM topic1;");
+    makeKsqlRequest("CREATE STREAM stream2 AS SELECT * FROM topic2;");
+    
+    // When
+    final CommandId commandId = new CommandId("TOPIC", "entity", "CREATE");
+    final Command command = new Command(
+        "statement",
+        Collections.emptyMap(),
+        Collections.emptyMap(),
+        Optional.empty(),
+        Optional.of(Command.VERSION + 1),
+        Command.VERSION + 1);
+
+    final Map<String, Object> kafkaProducerProperties = REST_APP.getKsqlRestConfig().getCommandProducerProperties();
+    kafkaProducerProperties.put(
+        ProducerConfig.TRANSACTIONAL_ID_CONFIG,
+        ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG)
+    );
+    kafkaProducerProperties.put(
+        ProducerConfig.ACKS_CONFIG,
+        "all"
+    );
+    final KafkaProducer<CommandId, Command> producer = new KafkaProducer<>(
+        kafkaProducerProperties,
+        InternalTopicSerdes.serializer(),
+        InternalTopicSerdes.serializer()
+    );
+
+    produceToCommandTopic(producer, commandId, command);
```

Comment:
Can all this be wrapped inside the `produceToCommandTopic`?

Reply:
done